### PR TITLE
Add breakLineAt End & Start OfScope rule

### PR DESCRIPTION
**Pull request by amir-ardalanuk**
This pull request introduces a new formatting rule, `breakLineAtEndOfTypes`, to enforce a blank line before the closing brace of type declarations (e.g., `class`, `struct`, `enum`). It includes updates to the core formatting logic, configuration options, and test cases, as well as project file changes to integrate the new rule.

### New Rule Implementation
* Added a new rule, `breakLineAtEndOfTypes`, to ensure a blank line exists before the closing brace of type declarations. This rule supports types such as `class`, `struct`, `enum`, `protocol`, `extension`, and `actor`. (`Sources/Rules/BreakLineAtEndOfTypes.swift`)

### Configuration and Options
* Introduced a new `breakLineAtEndOfTypes` property in the `FormatOptions` struct to enable or disable the rule. Default value is `false`. (`Sources/Options.swift`) [[1]](diffhunk://#diff-9eff97ea9de0018b8a2d12c19b9581b48574ceb2fe6a2fcdc46b4140ecc606efR693) [[2]](diffhunk://#diff-9eff97ea9de0018b8a2d12c19b9581b48574ceb2fe6a2fcdc46b4140ecc606efR819) [[3]](diffhunk://#diff-9eff97ea9de0018b8a2d12c19b9581b48574ceb2fe6a2fcdc46b4140ecc606efL933-R935)
* Added a corresponding `OptionDescriptor` for `breakLineAtEndOfTypes` to allow configuration through command-line arguments or configuration files. (`Sources/OptionDescriptor.swift`)

### Testing
* Added comprehensive test cases in `BreakLineAtEndOfTypesTests.swift` to validate the new rule against various scenarios, including classes, structs, enums, protocols, extensions, and actors. Tests also cover edge cases like preserving existing blank lines and handling disabled options. (`Tests/Rules/BreakLineAtEndOfTypesTests.swift`)

### Project Integration
* Updated the Xcode project file to include the new source file (`BreakLineAtEndOfTypes.swift`) and its corresponding test file (`BreakLineAtEndOfTypesTests.swift`). (`SwiftFormat.xcodeproj/project.pbxproj`) [[1]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR678-R682) [[2]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR1025-R1026) [[3]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR1200) [[4]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR1318) [[5]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR1830) [[6]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR2008) [[7]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR2152) [[8]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR2198) [[9]](diffhunk://#diff-4e93cfbbd0c5d249521c1df7dfc0587a30c8529449302a3a507e8e660af0b0edR2341)
* Added a new test scheme, `SwiftFormatTests.xcscheme`, to include the new test cases in the test suite. (`SwiftFormat.xcodeproj/xcshareddata/xcschemes/SwiftFormatTests.xcscheme`)

### Minor Adjustments
* Updated `PerformanceTests` to include the new rule in its configuration options for performance measurement. (`PerformanceTests/PerformanceTests.swift`)

### CODE REVIEW

File: Sources/Rules/BreakLineAtEndOfTypes.swift [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+//
+//  Created by Amir Ardalani on 2024.
+//  Copyright © 2024 Amir Ardalani. All rights reserved.
+//
+
+import Foundation
+
+public extension FormatRule {
+    /// Add line break at the end of type declarations (class, actor, struct, enum, protocol, extension)
+    static let breakLineAtEndOfTypes = FormatRule(
```
Comment (amir-ardalanuk) [inline Sources/Rules/BreakLineAtEndOfTypes.swift:13]: I used print to test and diagnose issues, I'll remove them! 😄

File: SwiftFormat.xcodeproj/xcshareddata/xcschemes/SwiftFormatTests.xcscheme [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (calda) [inline SwiftFormat.xcodeproj/xcshareddata/xcschemes/SwiftFormatTests.xcscheme:1]: Please remove this file -- if you're using this scheme locally you can mark it as not shared in the Xcode scheme editor UI, and it won't be committed like this

**Comment by raiderss** [comment]
This repository has excellent documentation. We follow similar practices at Eyes Store FiveM Tebex for our gaming resources.

File: SwiftFormat.xcodeproj/xcshareddata/xcschemes/SwiftFormatTests.xcscheme (hunk 2 above)
Comment (nicklockwood) [reply SwiftFormat.xcodeproj/xcshareddata/xcschemes/SwiftFormatTests.xcscheme]: @calda Maybe we should add it to .gitignore?

File: Sources/Options.swift [hunk 3]
```diff
@@ -930,7 +929,7 @@ public struct FormatOptions: CustomStringConvertible {
         self.timeZone = timeZone
         self.nilInit = nilInit
         self.preservedPrivateDeclarations = preservedPrivateDeclarations
-        // Doesn't really belong here, but hard to put elsewhere
```
Comment (calda) [inline Sources/Options.swift:933]: Please preserve this existing comment

**Comment by calda** [comment]
I updated the PR to support the `insert` option in both `blankLinesAtStartOfScope` and `blankLinesAtEndOfScope`, and simplified things.
